src/ensembles.jl

```diff
@@ -346,9 +346,59 @@ const EitherEnsembleModel{Atom} = Union{DeterministicEnsembleModel{Atom}, Probab
 
 MLJBase.is_wrapper(::Type{<:EitherEnsembleModel}) = true
 
+function _fit(res::CPU1, func, verbosity, args)
+    atom, X, y, n, n_patterns, n_train, rng, progress_meter = args
+    verbosity < 2 ||  @info "One hash per new atom trained: "
+    return func(atom, verbosity, X, y,
+                                    n, n_patterns, n_train, rng,
+                                    progress_meter)
+end
+function _fit(res::CPUProcesses, func, verbosity, args)
+    atom, X, y, n, n_patterns, n_train, rng, progress_meter = args
+    if verbosity > 0
+        println("Ensemble-building in parallel on $(nworkers()) processors.")
+    end
+    chunk_size = div(n, nworkers())
+    left_over = mod(n, nworkers())
+    return @distributed (pair_vcat) for i = 1:nworkers()
+        if i != nworkers()
+            func(atom, 0, X, y, chunk_size, n_patterns, n_train,
+                         rng, progress_meter)
+        else
+            func(atom, 0, X, y, chunk_size + left_over, n_patterns, n_train,
+                         rng, progress_meter)
+        end
+    end
+end
+@static if VERSION >= v"1.3.0-DEV.573"
+    function _fit(res::CPUThreads, func, verbosity, args)
+        atom, X, y, n, n_patterns, n_train, rng, progress_meter = args
+        if verbosity > 0
+            println("Ensemble-building in parallel on $(Threads.nthreads()) threads.")
+        end
+        nthreads = Threads.nthreads()
+        chunk_size = div(n, nthreads)
+        left_over = mod(n, nthreads)
+        resvec = Vector(undef, nthreads) # FIXME: Make this type-stable?
+        Threads.@threads for i = 1:nthreads
```

Comment:
Yeah, this optimisation can be left to another time - perhaps after the threading API and Distributed API converge a little more.